### PR TITLE
allow logging into a file with default logger

### PR DESCRIPTION
**Pull request by ekoby**

### CODE REVIEW

File: library/utils.c [hunk 1]
```diff
@@ -289,7 +289,16 @@ static void init_debug(uv_loop_t *loop) {
         tlsuv_set_debug(tlsuv_level, tlsuv_logger);
     }
 
-    ziti_debug_out = stderr;
+    char *logfile = getenv("ZITI_LOG_FILE");
+    if (logfile) {
```
Comment (dovholuknf) [inline library/utils.c:293]: i don't love that this doesn't cover log rotation type policies. it also opens the file but never seems to close it? will that leak the descriptor when this process exits or will it get cleaned up?

File: library/utils.c [hunk 2]
```diff
@@ -289,7 +289,16 @@ static void init_debug(uv_loop_t *loop) {
         tlsuv_set_debug(tlsuv_level, tlsuv_logger);
     }
 
-    ziti_debug_out = stderr;
+    char *logfile = getenv("ZITI_LOG_FILE");
```
Comment (scareything) [inline library/utils.c:292]: is there a reason for doing this when the log level is being set? it feels like this belongs in ziti_log_init with explicit API around it.

File: library/utils.c (hunk 1 above)
Comment (ekoby) [reply library/utils.c:293]: logrotation belongs outside of SDK, this is just an option for default logger. An application using SDK is free to override it

Comment (ekoby) [reply library/utils.c:293]: all process resources are cleanup up on exit

Comment (dovholuknf) [reply library/utils.c:293]: why provide it at all then? the sdk already has callback based functionality. seems like it should either be here and complete, or not and we let the consumer decide if it logs to  a file

File: library/utils.c (hunk 2 above)
Comment (ekoby) [reply library/utils.c:292]: this is just an option for default logger, see comment below

File: library/utils.c (hunk 1 above)
Comment (ekoby) [reply library/utils.c:293]: it is probably the next step (this is how it is done in other libraries -- openssl, etc) 
I want to keep this for now until downstream projects can implement it -- probably needs zitilib layer API

**Pull request closed without merging**